Source/core/Portability.h

```diff
@@ -787,6 +794,23 @@ namespace Core {
     const uint32_t infinite = -1;
     static const string emptyString;
 
+
+    class Demangling {
+    public:
+        static Demangling& demanglingInstance() {static Demangling demangleClassNames; return demangleClassNames;}
+
+    private:
+        Demangling() = default;
+        ~Demangling() = default;
+        Demangling(const Demangling&) = delete;
+        Demangling& operator=(const Demangling&) = delete;
+
+    public:
+        TextFragment Demangled(const char name[]);
+        TextFragment ClassName(const char name[]);
+        TextFragment ClassNameOnly(const char name[]);
+    };
+
```

Comment:
As ClassName/ClassNameOnly and Demangled are now methods on a class with *no* memebers, why not just create methods in a namespace Demangling ? Something like:
`
namespace Thunder { namespace Core { namespace Demangling {
    TextFragment Demangled(const char name[]);
    TextFragment ClassName(const char name[]);
    TextFragment ClassNameOnly(const char name[]);
}}}
`

This makes the usage simpler and simpler...